serverless-single-page-app-plugin/index.js

```diff
@@ -0,0 +1,144 @@
+"use strict";
+
+const spawnSync = require("child_process").spawnSync;
+
+class ServerlessPlugin {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.commands = {
+      syncToS3: {
+        usage: "Deploys the `app` directory to your bucket",
+        lifecycleEvents: ["sync"],
+      },
+      domainInfo: {
+        usage: "Fetches and prints out the deployed CloudFront domain names",
+        lifecycleEvents: ["domainInfo"],
+      },
+      invalidateCloudFrontCache: {
+        usage: "Invalidates CloudFront cache",
+        lifecycleEvents: ["invalidateCache"],
+      },
+    };
+
+    this.hooks = {
+      "syncToS3:sync": this.syncDirectory.bind(this),
+      "domainInfo:domainInfo": this.domainInfo.bind(this),
+      "invalidateCloudFrontCache:invalidateCache":
+        this.invalidateCache.bind(this),
+    };
+  }
+
+  runAwsCommand(args) {
+    let command = "aws";
+    if (this.serverless.variables.service.provider.region) {
+      command = `${command} --region ${this.serverless.variables.service.provider.region}`;
+    }
+    if (this.serverless.variables.service.provider.profile) {
+      command = `${command} --profile ${this.serverless.variables.service.provider.profile}`;
+    }
+    const result = spawnSync(command, args);
```

Comment:
Suppose you might have some troubles with invalidation action. Please re-check. A possible solution is available at https://github.com/serverless/examples/issues/510

Reply:
Thank you for your review. During the auto-deploy, I didn't have the described problems, but I will necessarily double-check everything.